src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,180 @@
+    ///         Release | AnyCPU | net46
+    ///         
+    ///     -> Say, active VS configuration = "Debug | AnyCPU"
+    ///       
+    ///     -> Active configurations ignoring TargetFramework returned by this provider:
+    ///         Debug | AnyCPU | net45
+    ///         Debug | AnyCPU | net46
+    /// </summary>
+    [Export(typeof(ActiveConfiguredProjectsIgnoringTargetFrameworkProvider))]
+    internal class ActiveConfiguredProjectsIgnoringTargetFrameworkProvider : OnceInitializedOnceDisposedAsync
+    {
+        private readonly object _gate = new object();
+        private readonly IUnconfiguredProjectServices _services;
+        private readonly IUnconfiguredProjectCommonServices _commonServices;
+        private readonly IProjectAsynchronousTasksService _tasksService;
+        private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
+
+        // Cache the last queried TargetFrameworks and associated configured projects per target framework map.
+        // Read/writes for both these fields must be done within a lock to keep them in sync.
+        private ImmutableDictionary<string, ConfiguredProject> _cachedConfiguredProjectsMap;
+        private string _cachedTargetFrameworks;
+
+        private IDisposable _evaluationSubscriptionLink;
+
+        [ImportingConstructor]
+        public ActiveConfiguredProjectsIgnoringTargetFrameworkProvider(
+            IUnconfiguredProjectServices services,
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService)
+            : base(commonServices.ThreadingService.JoinableTaskContext)
+        {
+            Requires.NotNull(services, nameof(services));
+            Requires.NotNull(commonServices, nameof(commonServices));
+            Requires.NotNull(tasksService, nameof(tasksService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+
+            _services = services;
+            _commonServices = commonServices;
+            _tasksService = tasksService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+        }
+
+        [ProjectAutoLoad(startAfter: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal async Task OnProjectFactoryCompletedAsync()
+        {
+            await InitializeCoreAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {
+            using (_tasksService.LoadedProject())
+            {
+                await GetActiveConfiguredProjectsMapAsync().ConfigureAwait(false);
```

Comment:
What exactly is the point of this call? It "primes" the _cachedConfiguredProjectMap?